src/bg/on-user-script-xhr.js

```diff
@@ -119,4 +127,37 @@ function open(xhr, d, port) {
   }
 }
 
+function getHeader(headers, name) {
+  name = name.toLowerCase();
+  for (var header of headers) {
+    if (header.name.toLowerCase() === name) {
+      return header;
+    }
+  }
+  return null;
+}
+
+var extensionUrl = browser.extension.getURL("");
+
+function rewriteHeaders(e) {
+  if (e.originUrl && e.originUrl.startsWith(extensionUrl)) {
+    for (var name of headersToReplace) {
+      var prefixedHeader = getHeader(e.requestHeaders, dummyHeaderPrefix + name);
+      if (prefixedHeader) {
+        if (!getHeader(e.requestHeaders, name)) {
+          // only try to add real header if request doesn't already have it
```

Comment:
I'm not sure I understand the purpose of this conditional, could you provide an example of when `getHeader` may return true on line 147?

Further, under what condition would you _not_ want to replace a header with the prefixed version?

Reply:
It would return true for headers like `Accept-Encoding` and `Connection` that are always automatically set, if we were allowing the user to replace those.

>under what condition would you not want to replace a header with the prefixed version?

I guess you wouldn't.